### ISSUE TITLE
missing cmdlets with powershell 7

### ISSUE THREAD

**Issue by danpousson** [dataplat/dbatools#8768]
### Verified issue does not already exist?

I have searched and found no existing issue

### What error did you receive?

most modules are not present:

example:
[no errorinvoke-dbaquery: The term 'invoke-dbaquery' is not recognized as a name of a cmdlet, function, script file, or executable program.](invoke-dbaquery: The term 'invoke-dbaquery' is not recognized as a name of a cmdlet, function, script file, or executable program.)

### Steps to Reproduce

When I install dbatools on powershell 7, I only get these modules below:

get-command -Module dbatools | select Name
Attach-DbaDatabase
Detach-DbaDatabase
Get-DbaBuildReference
Get-DbaDbModule
Get-DbaRegisteredServer
Start-SqlMigration
Write-DbaDataTable
Select-DbaObject
Set-DbatoolsConfig

Is there a way to import all supported modules?
it looks like i see them [673 items] in: C:\Program Files\PowerShell\Modules\dbatools\1.1.145\functions

powershell install method: iex "& { $(irm https://aka.ms/install-powershell.ps1) } -UseMSI"
dbatools versions tried: 1.1.141 - latest pre-release (i one by one removed and installed different versions)

- I have local admin
- I am running pwsh as Administator
- I have tried this with -scope AllUsers and current user

### Please confirm that you are running the most recent version of dbatools

yes: [1.1.145 (current version)](https://www.powershellgallery.com/packages/dbatools/1.1.145)

### What PowerShell host was used when producing this error

PowerShell Core (pwsh.exe)

### PowerShell Host Version

Name                           Value
----                           -----
PSVersion                      7.3.2
PSEdition                      Core
GitCommitId                    7.3.2
OS                             Microsoft Windows 10.0.17763
Platform                       Win32NT
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0…}
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
WSManStackVersion              3.0

### SQL Server Edition and Build number

n/a

### .NET Framework Version

.NET 7.0.2

**Comment by danpousson** [comment]
what modules should be available?

**Comment by wsmelton** [comment]
Once you import the module, all commands that are available to you can be found using:

```powershell
Get-Command -Module dbatools
```

**Comment by danpousson** [comment]
all i see are these:


get-command -Module dbatools | select Name
Attach-DbaDatabase
Detach-DbaDatabase
Get-DbaBuildReference
Get-DbaDbModule
Get-DbaRegisteredServer
Start-SqlMigration
Write-DbaDataTable
Select-DbaObject
Set-DbatoolsConfig

**Comment by wsmelton** [comment]
The module did not import correctly or was prevented from importing completely.

**Comment by danpousson** [comment]
thank you, i will investigate those last 2 comments

**Comment by wsmelton** [comment]
If the import was completely successful there are 662 commands total available (at least on PowerShell 7.3.2 on Windows 10 OS)

![image](https://user-images.githubusercontent.com/11204251/220215021-d60ce670-3aad-4edb-96be-1277451388c4.png)

**Comment by danpousson** [comment]
so werid:

PS C:\Program Files\PowerShell\Modules\dbatools\1.1.145> import-module -name dbatools  -Verbose
```
VERBOSE: Loading module from path 'C:\Program Files\PowerShell\Modules\dbatools\1.1.145\dbatools.psd1'.
VERBOSE: Importing cmdlet 'Select-DbaObject'.
VERBOSE: Importing cmdlet 'Set-DbatoolsConfig'.
VERBOSE: Importing alias 'Attach-DbaDatabase'.
VERBOSE: Importing alias 'Detach-DbaDatabase'.
VERBOSE: Importing alias 'Get-DbaBuildReference'.
VERBOSE: Importing alias 'Get-DbaDbModule'.
VERBOSE: Importing alias 'Get-DbaRegisteredServer'.
VERBOSE: Importing alias 'Start-SqlMigration'.
VERBOSE: Importing alias 'Write-DbaDataTable'.
```
PS C:\Program Files\PowerShell\Modules\dbatools\1.1.145>

**Comment by wsmelton** [comment]
Delete the version and reinstall from the PowerShell Gallery

**Comment by danpousson** [comment]
I'm able to recreate the issue in ps5.1 when adding -verbose to the import command as mentioned above. 

Looks like AV maybe be blocking allcommands.ps1

looks like this file may be getting blocked:
`This script contains malicious content and has been blocked by your antivirus software:`
allcommands.ps1
C:\Program Files\PowerShell\Modules\dbatools\1.1.145\

&

C:\Program Files\WindowsPowerShell\Modules\dbatools\1.1.145\allcommands.ps1 
that is my current path but it could vary based on install

**Comment by wsmelton** [comment]
What AV are you using? We have another issue for one that don't think the AV software is going to fix.

**Comment by danpousson** [comment]
too many:
Carbon Black (CB Defense)
Cylance Protect

I can ask them which one blocked it if you want.

**Comment by wsmelton** [comment]
I can already tell you it will be Carbon Black.

Going to start tracking this in #8241 and I've pinned this issue so more users can be made aware.

**Comment by danpousson** [comment]
update: FYI a Carbon Black agent update plus reboot fixed this for 5.1 & pwsh 7
AV issue

**Comment by potatoqualitee** [comment]
Please try with the latest version, too. dbatools is now signed with a Microsoft-issued certificate which has immediate reputation and I don't think we'll have anymore AV issues. You can read more here: https://blog.netnerds.net/2025/08/dbatools-azure-trusted-signing/
